lua/neo-tree/ui/popups.lua

```diff
@@ -19,11 +19,16 @@ M.popup_options = function(title, min_width, override_options)
   local nt = require("neo-tree")
   local popup_border_style = nt.config.popup_border_style
   local popup_border_text = NuiText(" " .. title .. " ", highlights.FLOAT_TITLE)
+  local col = 0
+  local offset = (vim.api.nvim_win_get_width(0) - width) - 2
+  if vim.api.nvim_win_get_number(0) ~= 1 and offset < 0 then
```

Comment:
I don't think this is always going to work out the way you want it to. A window could still be on the left and not be window number 1.

Reply:
I've tried some quick test to force this and check the docs `(windows are numbered from top-left to bottom-right)`. So I guessed it could be a save way.

Do you have a suggestion how to solve this? I mistakenly though pcalling the position would fix the error with floating windows so I removed it from the comment above.

Reply:
If you look at this screenshot, you'll see that Neo-tree is actually window 2 even though it is left aligned.
![image](https://user-images.githubusercontent.com/5160605/179640994-f729a126-8fc2-4497-b1e9-43497c41ad3e.png)

I think the correct solution is to just calculate the starting column for the window as an absolute position within the neovim editor, and see if the width of the popup will end up beyond the max columns (`vim.o.columns`). If so, then move the position of the popup whatever amount is needed to keep it within the editor window. This way it will always work regardless of the position of the parent window.